community/security/src/main/java/org/neo4j/server/security/auth/UserManager.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.server.security.auth;
+
+import java.io.IOException;
+
+import org.neo4j.server.security.auth.exception.IllegalUsernameException;
+
+public interface UserManager
+{
+    User newUser( String username, String initialPassword, boolean requirePasswordChange ) throws IOException,
+            IllegalUsernameException;
+
+    boolean deleteUser( String username ) throws IOException;
+
+    User getUser( String username );
+
+    User setUserPassword( String username, String password ) throws IOException;
```

Comment:
See my other comment on this method, but if we keep it like this we should at least document what a `null` return means here.